src/Table/Table.tsx

```diff
@@ -0,0 +1,977 @@
+const defaultSort: JSX.Element = <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 644"><path transform="translate(0 240)" d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z" /><path transform="translate(0 -100)" d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z" /></svg>;
+
+export type DataItem = { [name: string]: any } & TableRow;
+
+export interface Column {
+    label: string;
+    accessor: string;
+    canSort?: boolean;
+}
+
+export interface ActionLinkItem {
+    label: string;
+    onClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, selectedRow: TableRow) => void;
+}
+
+export interface PrimaryActionButton {
+    label: string;
+    onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, selectedRow: TableRow) => void;
+}
+
+export interface TableHeader extends Column {
+    isSorted?: boolean;
+    isSortedDesc?: boolean;
+}
+
+interface Cell {
+    id: string | number;
+    accessor: string;
+    value: string | number | boolean;
+}
+
+export interface TableRow {
+    rowIndex: number;
+    cells: Array<Cell>;
+    selected?: boolean;
+    actionsDropdownDropped?: boolean;
+    subRows?: Array<TableRow>;
+    expanded?: boolean;
+    rowContentDetail?: React.ReactNode;
+}
+
+export const enum sortDirectionTypes {
+    Ascending = "ASC",
+    Descending = "DESC"
+}
+
+/**
+ * sum the total of columns or cols in a row
+ * @param colsLength the length of the columns
+ * @param useSelection add a column for selection checkboxes
+ * @param useShowActionColumn add another column for action columns
+ * @param useGroupBy add another columns for groupby
+ */
+function sumCols(colsLength: number, useSelection?: boolean, useShowActionColumn?: boolean, useGroupBy?: boolean): number {
+    let sum = colsLength;
+    if (useSelection || useGroupBy) {
```

Comment:
Can this be written like this
```suggestion
    if (useSelection || useGroupBy || useShowActionColumn) { 
    sum = sum+1;
    }
```

Reply:
No it cannot be, They are independent and each has its bearing.

Reply:
for example if i call sumCols function with this parameters:
```
sumCols(colsLength = 3, useSelection = true, useShowActionColumn  = true, useGroupBy  = true)
sumCols(colsLength = 3, useSelection = false, useShowActionColumn  = true, useGroupBy  = true)
```
What is the expected return value for each case

Reply:
Plus one for every true